wafv2_aws_managed_rules/variables.tf

```diff
@@ -0,0 +1,41 @@
+variable "log_destinations" {
+  description = "A list of destinations to send the waf logs to"
+  type        = list(string)
+  default     = []
+}
+
+variable "arn_associations" {
+  description = "List of AWS resource ARNs for WAF rule association"
+}
+
+variable "aws_managed_rules" {
+  description = "List of AWS managed rules to be applied"
+  type        = list(any)
```

Comment:
You can get it to validate the type by being more specific with the type. Something like
```
object({
    name = string
    priority = number
    managed_rule_group_statement_name = string
    managed_rule_group_statement_vendor_name = string
    metric_name = string
  })
```
That way, terraform will complain if you don't put the right fields in. You can make some of them optional by doing `metric_name = optional(string, "")` if they're not all needed.